rotkehlchen/exchanges/binance.py

```diff
@@ -589,6 +607,140 @@ def _query_lending_balances(
+                    except (UnsupportedAsset, UnknownAsset) as e:
+                        error_type = 'unknown' if isinstance(e, UnknownAsset) else 'unsupported'
+                        self.msg_aggregator.add_warning(
+                            f'Found {error_type} {self.name} asset {e.identifier}. '
+                            f'Ignoring its lending interest history query.',
+                        )
+                        continue
+                    except KeyError as e:
+                        self.msg_aggregator.add_error(
+                            f'Missing key entry for {str(e)} in {self.name} {entry}. '
+                            f'Ignoring its lending interest history query.',
+                        )
+                    except DeserializationError as e:
+                        self.msg_aggregator.add_error(
+                            f'Error at deserializing {self.name} asset. {str(e)}. '
+                            f'Ignoring its lending interest history query.',
+                        )
+                        continue
+
+            if len(events) != 0:
+                with self.db.user_write() as write_cursor:
+                    try:
+                        self.history_events_db.add_history_events(write_cursor, events)
+                    except InputError as e:
+                        self.msg_aggregator.add_error(
+                            f'Failed to save Binance lending interest history {query_start_ts} '
+                            f'to {query_end_ts} in database. {str(e)}',
+                        )
+
+                    ranges.update_used_query_range(
+                        write_cursor=write_cursor,
+                        location_string=range_query_name,
+                        queried_ranges=[(start_ts, end_ts)] + ranges_to_query,
+                    )
+
+    def query_lending_interests_history_balance(
+            self,
+    ) -> dict[str, defaultdict[AssetWithOracles, Balance]]:
+        """Queries binance lending interest history and aggregate the interest gained per asset.
+        May raise:
+        - RemoteError
+
+        https://binance-docs.github.io/apidocs/spot/en/#get-interest-history-user_data-2
+        """
+        balances: defaultdict[AssetWithOracles, Balance] = defaultdict(Balance)
+        interests: defaultdict[AssetWithOracles, Balance] = defaultdict(Balance)
+        now = ts_now()
+        with self.db.conn.read_ctx() as cursor:
+            self._query_lending_interests_history(
+                cursor=cursor,
+                start_ts=BINANCE_LAUNCH_TS,
+                end_ts=now,
+            )
+
+            filter_query = HistoryEventFilterQuery.make(
+                from_ts=Timestamp(0),
+                to_ts=now,
+                event_types=[HistoryEventType.RECEIVE],
+                event_subtypes=[HistoryEventSubType.REWARD],
+                location=Location.BINANCE,
+                location_label=self.name,
+            )
```

Comment:
It doesn't matter much but since this doesn't use the filter it can be created outside the with statement